public/js/toc.js

```diff
@@ -0,0 +1,125 @@
+//
+//  toc.js
+//
+//  - Manages the table of contents
+//
+//  Copyright (c) 2015 Visionist, Inc.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+function TOC(editor){
+
+  this.editor = editor;
+  this.editorText = this.editor.getValue();
+  this.rootId = "toc-root";
+  this.level = 0;
+  this.lines = [];
+  this.heads = [];
+
+  /*
+  *  Parses each line and checks to see if the line is a header. If it is a header,
+  *  it checks to see what level
+  */
+   this.parse = function(){
+
+    this.lines = this.editorText.split('\n');
+
+
+    //Generate regexp for different header levels
+    var lvl1Reg = new RegExp(/^#{1}/);
+    var lvl2Reg = new RegExp(/^#{2}/);
+    var lvl3Reg = new RegExp(/^#{3}/);
+    var lvl4Reg = new RegExp(/^#{4}/);
+    var lvl5Reg = new RegExp(/^#{5}/);
+    var lvl6Reg = new RegExp(/^#{6}/);
+
```

Comment:
Is it possible to use one regular expression that looks for one or more `#` at the beginning of the line, captures that, and then uses the length?  This might be cleaner and a little bit more practical than having the 6 different ones, and testing each one.